bulk_site_creator/views.py

```diff
@@ -169,7 +198,11 @@ def new_job(request):
 @login_required
 @lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
 @require_http_methods(["POST"])
-def create_bulk_job(request: HttpRequest) -> Optional[JsonResponse]:
+def create_bulk_job(request: HttpRequest) -> HttpResponseRedirect:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 21 other functions. By comparison, 99% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 258-317 into a separate function.